src/main/java/org/victorrobotics/frc/dtlib/drivetrain/DTAccelerationLimit.java

```diff
@@ -0,0 +1,59 @@
+package org.victorrobotics.frc.dtlib.drivetrain;
+
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+
+public final class DTAccelerationLimit {
+    public final double maxAccelTranslation;
+    public final double maxAccelRotation;
+
+    public DTAccelerationLimit() {
+        maxAccelTranslation = Double.NaN;
+        maxAccelRotation = Double.NaN;
+    }
+
+    public DTAccelerationLimit(double translation, double rotation, double cycleLength) {
+        maxAccelTranslation = Double.isFinite(translation) ? Math.abs(translation * cycleLength)
+                : Double.NaN;
+        maxAccelRotation = Double.isFinite(rotation) ? Math.abs(rotation * cycleLength)
+                : Double.NaN;
+    }
+
+    public boolean apply(ChassisSpeeds newSpeeds, ChassisSpeeds previousSpeeds) {
+        boolean changed = false;
+
+        if (!Double.isNaN(maxAccelTranslation)) {
+            Translation2d oldTranslation = new Translation2d(previousSpeeds.vxMetersPerSecond,
+                    previousSpeeds.vyMetersPerSecond);
+            Translation2d newTranslation = new Translation2d(newSpeeds.vxMetersPerSecond,
+                    newSpeeds.vyMetersPerSecond);
+            Translation2d translationAccel = newTranslation.minus(oldTranslation);
+            double translationAccelMagnitude = translationAccel.getNorm();
```

Comment:
`translationAccelMagnitude` is `dv`, we should calculate `dv/dt` here so that accelerations are in "normal" units (m/s^2).

Reply:
Perhaps as an additional calculation, but for additional speed the conversion from cycles to seconds and back was skipped. Maybe just a comment to clarify? This is an internal implementation, so I'm not too concerned with this unless you see it as a code readability issue.

Reply:
I see, you're translating `maxAccelTranslation` into m/s/cycle in the constructor... I would prefer that all code be written to use standardized units. If anyone comes into this code and adds logging or looks at values during debugging it could be confusing.

Reply:
I’ll add some comments explaining the unit choice, and save the acceleration in standard units to the object as well.